gslb/ingestion/event_handlers.go

```diff
@@ -460,6 +484,66 @@ func AddNamespaceEventHandler(numWorkers uint32, c *GSLBMemberController) cache.
 				return
 			}
 			if oldNS.ResourceVersion != ns.ResourceVersion {
+				oldTenant, oldExists := oldNS.Annotations[gslbutils.TenantAnnotation]
+				if !oldExists {
+					oldTenant = gslbutils.GetTenant()
+				}
+				newTenant, newExists := ns.Annotations[gslbutils.TenantAnnotation]
+				if !newExists {
+					newTenant = gslbutils.GetTenant()
+				}
+				if oldTenant != newTenant {
```

Comment:
we don't need to check tenant for hostrule since we will determine tenant via ingress/route. Have tested this scenario of tenant update with hostrule.